From 51c425d1050ebde6566da8a4ed248640233330ae Mon Sep 17 00:00:00 2001
From: Amelia Crate <amelia.crate@chainguard.dev>
Date: Wed, 10 Dec 2025 20:54:48 +0000
Subject: [PATCH] Add compiler flags for compatibility with objtool's orc,
 rethunk, and sls validation.

See https://github.com/NVIDIA/open-gpu-kernel-modules/issues/429

I do not know why this was never upstreamed.
---
 src/nvidia-modeset/Makefile | 3 +++
 src/nvidia/Makefile         | 2 ++
 2 files changed, 5 insertions(+)

diff --git a/src/nvidia-modeset/Makefile b/src/nvidia-modeset/Makefile
index b54138cc..9f5d07a8 100644
--- a/src/nvidia-modeset/Makefile
+++ b/src/nvidia-modeset/Makefile
@@ -155,6 +155,9 @@ ifeq ($(TARGET_ARCH),x86_64)
   CONDITIONAL_CFLAGS += $(call TEST_CC_ARG, -fno-jump-tables)
   CONDITIONAL_CFLAGS += $(call TEST_CC_ARG, -mindirect-branch=thunk-extern)
   CONDITIONAL_CFLAGS += $(call TEST_CC_ARG, -mindirect-branch-register)
+  CONDITIONAL_CFLAGS += $(call TEST_CC_ARG, -fno-asynchronous-unwind-tables)
+  CONDITIONAL_CFLAGS += $(call TEST_CC_ARG, -mharden-sls=all)
+  CONDITIONAL_CFLAGS += $(call TEST_CC_ARG, -mfunction-return=thunk-extern)
 endif
 
 CFLAGS += $(CONDITIONAL_CFLAGS)
diff --git a/src/nvidia/Makefile b/src/nvidia/Makefile
index d1d6d866..213e8030 100644
--- a/src/nvidia/Makefile
+++ b/src/nvidia/Makefile
@@ -185,6 +185,8 @@ ifeq ($(TARGET_ARCH),x86_64)
   CONDITIONAL_CFLAGS += $(call TEST_CC_ARG, -fno-jump-tables)
   CONDITIONAL_CFLAGS += $(call TEST_CC_ARG, -mindirect-branch-register)
     CONDITIONAL_CFLAGS += $(call TEST_CC_ARG, -mindirect-branch=thunk-extern)
+  CONDITIONAL_CFLAGS +=$(call TEST_CC_ARG, -mharden-sls=all)
+  CONDITIONAL_CFLAGS +=$(call TEST_CC_ARG, -mfunction-return=thunk-extern)
 endif
 
 CFLAGS += $(CONDITIONAL_CFLAGS)
-- 
2.50.1

